azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/functions/CleanupResources.java

```diff
@@ -58,36 +69,67 @@ public CleanupResources(AzureComputeApi azureComputeApi,
+            boolean deploymentDeleteStatus = false;
+
+            if (jobDone) {
+               StorageService ss = api.getStorageAccountApi(group).get(storageAccountName);
+               if (ss != null) {
+                  storageAcctDeleteStatus = api.getStorageAccountApi(group).delete(storageAccountName);
+               } else {
+                  storageAcctDeleteStatus = true;
+               }
+               Deployment deployment = api.getDeploymentApi(group).get(id);
+               if (deployment != null) {
                   uri = api.getDeploymentApi(group).delete(id);
                   jobDone = resourceDeleted.apply(uri);
                   if (jobDone) {
-                     // Delete public ip
-                     boolean ipDeleteStatus = api.getPublicIPAddressApi(group).delete(id + "publicip");
-
-                     // Delete Virtual network
-                     boolean vnetDeleteStatus = api.getVirtualNetworkApi(group).delete(group + "virtualnetwork");
-                     return ipDeleteStatus && vnetDeleteStatus;
-
+                     deploymentDeleteStatus = true;
+                  }
+               } else {
+                  deploymentDeleteStatus = true;
+               }
+               NetworkInterfaceCard nic = api.getNetworkInterfaceCardApi(group).get(id + "nic");
+               if (nic != null) {
+                  uri = api.getNetworkInterfaceCardApi(group).delete(id + "nic");
+                  if (uri != null) {
+                     jobDone = resourceDeleted.apply(uri);
+                     if (jobDone) {
+                        boolean ipDeleteStatus = false;
+                        PublicIPAddress ip = api.getPublicIPAddressApi(group).get(id + "publicip");
+                        if (ip != null) {
+                           ipDeleteStatus = api.getPublicIPAddressApi(group).delete(id + "publicip");
+                        } else {
+                           ipDeleteStatus = true;
+                        }
+
+                        // Get NSG
+                        boolean nsgDeleteStatus = false;
+                        NetworkSecurityGroup nsg = api.getNetworkSecurityGroupApi(group).get(id + "nsg");
+                        if (nsg != null) {
+                           uri = api.getNetworkSecurityGroupApi(group).delete(id + "nsg");
+                           jobDone = resourceDeleted.apply(uri);
+                           if (jobDone) {
+                              nsgDeleteStatus = true;
+
+                           }
+                        }
+                        else {
+                           nsgDeleteStatus = true;
+                        }
+
+                        return deploymentDeleteStatus && storageAcctDeleteStatus && ipDeleteStatus && nsgDeleteStatus;
+                     } else {
+                        return false;
+                     }
                   } else {
                      return false;
```

Comment:
@nacx @andreaturli If you guys don't mind, let's get this PR done first. Then we will add this create with tag and cleanup by tag feature in a subsequent PR. 

As for adding a tag with the id of the VM, I recommend not because some of these resources are shared between nodes, for example vnet and subnet.